wormhole-connect/src/views/v2/Bridge/Routes/index.tsx

```diff
@@ -52,11 +65,70 @@ const Routes = (props: Props) => {
     return routeStates.filter((rs) => rs.supported);
   }, [routeStates]);
 
+  const supportedRoutesNames = useMemo(
+    () => supportedRoutes.map((r) => r.name),
+    [supportedRoutes],
+  );
+
+  const { quotesMap, isFetching } = useRoutesQuotesBulk(supportedRoutesNames, {
+    amount: Number.parseFloat(amount),
+    sourceChain: fromChain,
+    sourceToken: token,
+    destChain: toChain,
+    destToken,
+    nativeGas: toNativeToken,
+  });
+
   const walletsConnected = useMemo(
     () => !!sendingWallet.address && !!receivingWallet.address,
     [sendingWallet.address, receivingWallet.address],
   );
 
+  const sortedSupportedRoutes = useMemo(
+    () =>
+      [...supportedRoutes].sort((routeA, routeB) => {
+        const quoteA = quotesMap[routeA.name];
+        const quoteB = quotesMap[routeB.name];
+        const routeConfigA = config.routes.get(routeA.name);
+        const routeConfigB = config.routes.get(routeB.name);
+
+        // 1. Prioritize automatic routes
+        if (routeConfigA.AUTOMATIC_DEPOSIT && !routeConfigB.AUTOMATIC_DEPOSIT) {
+          return -1;
+        } else if (
+          !routeConfigA.AUTOMATIC_DEPOSIT &&
+          routeConfigB.AUTOMATIC_DEPOSIT
+        ) {
+          return 1;
+        }
+
+        if (quoteA && quoteB) {
+          // 2. Prioritize estimated time
+          if (quoteA.eta > quoteB.eta) {
+            return 1;
+          } else if (quoteA.eta < quoteB.eta) {
+            return -1;
+          }
+
+          // 3. Compare relay fees
+          if (quoteA.relayerFee > quoteB.relayerFee) {
+            return 1;
+          } else if (quoteA.relayerFee < quoteB.relayerFee) {
+            return -1;
+          }
+        }
+
+        // Don't swap when routes match by all criteria or don't have quotas
```

Comment:
If one has a quote and the other doesn't, shouldn't we prefer the one that does?

Reply:
Added final priority, routes with a quote over w/o one.